theatre/studio/src/panels/SequenceEditorPanel/DopeSheet/Right/BasicKeyframedTrack/KeyframeEditor/CurveEditorPopover/shared.ts

```diff
@@ -0,0 +1,89 @@
+  x1: number,
+  y1: number,
+  x2: number,
+  y2: number,
+]
+
+export function cssCubicBezierArgsFromHandles(points: CubicBezierHandles) {
+  return points.map((p) => p.toFixed(2)).join(', ')
+}
+
+const MAX_REASONABLE_BEZIER_STRING = 128
+export function handlesFromCssCubicBezierArgs(
+  str: string | undefined | null,
+): null | CubicBezierHandles {
+  if (!str || str?.length > MAX_REASONABLE_BEZIER_STRING) return null
+  const args = str.split(',')
+  if (args.length !== 4) return null
+  const nums = args.map((arg) => {
+    return Number(arg.trim())
+  })
+
+  if (!nums.every((v) => isFinite(v))) return null
+
+  if (nums[0] < 0 || nums[0] > 1 || nums[2] < 0 || nums[2] > 1) return null
+  return nums as CubicBezierHandles
+}
+
+/**
+ * A collection of cubic-bezier approximations of common easing functions
+ * ref: https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function
+ **/
+export const EASING_PRESETS = [
+  {label: 'Linear', value: '0.5, 0.5, 0.5, 0.5'},
+  {label: 'Sine In', value: '0.470, 0, 0.745, 0.715'},
+  {label: 'Sine Out', value: '0.390, 0.575, 0.565, 1'},
+  {label: 'Sine In Out', value: '0.445, 0.050, 0.550, 0.950'},
+
+  {label: 'Cubic In', value: '0.550, 0.055, 0.675, 0.190'},
+  {label: 'Quart In', value: '0.895, 0.030, 0.685, 0.220'},
+  {label: 'Quint In', value: '0.755, 0.050, 0.855, 0.060'},
+  {label: 'Expo In', value: '0.78, 0.00, 0.81, 0.00'},
+
+  {label: 'Cubic Out', value: '0.215, 0.610, 0.355, 1'},
+  {label: 'Quart Out', value: '0.165, 0.840, 0.440, 1'},
+  {label: 'Quint Out', value: '0.230, 1, 0.320, 1'},
+  {label: 'Expo Out', value: '0.190, 1, 0.220, 1'},
+
+  {label: 'Cubic In Out', value: '0.645, 0.045, 0.355, 1'},
+  {label: 'Quart In Out', value: '0.770, 0, 0.175, 1'},
+  {label: 'Quint In Out', value: '0.860, 0, 0.070, 1'},
+  {label: 'Expo In Out', value: '1, 0, 0, 1'},
+
+  /* These easings are not being included initially in order to
+     simplify the choices */
+  // {label: 'Back In Out', value: '0.680, -0.550, 0.265, 1.550'},
+  // {label: 'Back In', value: '0.600, -0.280, 0.735, 0.045'},
+  // {label: 'Back Out', value: '0.175, 0.885, 0.320, 1.275'},
+
+  // {label: 'Circ In Out', value: '0.785, 0.135, 0.150, 0.860'},
+  // {label: 'Circ In', value: '0.600, 0.040, 0.980, 0.335'},
+  // {label: 'Circ Out', value: '0.075, 0.820, 0.165, 1'},
+
+  // {label: 'Quad In Out', value: '0.455, 0.030, 0.515, 0.955'},
+  // {label: 'Quad In', value: '0.550, 0.085, 0.680, 0.530'},
+  // {label: 'Quad Out', value: '0.250, 0.460, 0.450, 0.940'},
+
+  // {label: 'Ease Out In', value: '.42, 0, .58, 1'},
+]
+
+const SIMILARITY_CUTOFF = 0.02
+export function areEasingsSimilar(
+  easing1: CubicBezierHandles | null | undefined,
+  easing2: CubicBezierHandles | null | undefined,
+) {
+  if (!easing1 || !easing2) return false
+  let totalDiff = 0
+  for (let i = 0; i < 4; i++) {
+    totalDiff += Math.abs(easing1[i] - easing2[i])
+  }
+  return totalDiff < SIMILARITY_CUTOFF
+}
```

Comment:
Change: Add doc comment about how this is being used. Potentially require that the similarity cutoff is provided to the function via an options object. This way, two different users of this `areEasingsSimilar` don't need to worry about tweaking `SIMILARITY_CUTOFF` constant and affecting the other.

4/10: Using an options object, is nice as it requires the user-code to explicitly name the argument it's passing in so it's easier to read for others later on.

Reply:
Done